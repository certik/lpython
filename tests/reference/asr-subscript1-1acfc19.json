{
    "basename": "asr-subscript1-1acfc19",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/subscript1.py",
    "infile_hash": "5d229893d3e13ea4463e8ed47eb3798be0b8c28f5ef6b6c773e87b80",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-subscript1-1acfc19.stdout",
    "stdout_hash": "5464d1192f8e11d05dcd3172a28c5710c79bcac6cb92aa3193bc2e48",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}