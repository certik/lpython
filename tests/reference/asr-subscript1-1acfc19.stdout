(TranslationUnit (SymbolTable 1 {main_program: (Program (SymbolTable 3 {}) main_program [] []), test_subscript: (Subroutine (SymbolTable 2 {A: (Variable 2 A Local () () Default (Integer 4 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 5 (Integer 4 [])))]) Source Public Required .false.), B: (Variable 2 B Local () () Default (Integer 4 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 2 (Integer 4 [])))]) Source Public Required .false.), i: (Variable 2 i Local () () Default (Integer 4 []) Source Public Required .false.), s: (Variable 2 s Local () () Default (Character 1 -2 () []) Source Public Required .false.)}) test_subscript [] [(= (Var 2 s) (ConstantString "abc" (Character 1 3 () [])) ()) (= (Var 2 s) (ArrayRef 2 s [((BinOp (ConstantInteger 0 (Integer 4 [])) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()) (BinOp (ConstantInteger 0 (Integer 4 [])) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()) ())] (Character 1 -2 () []) ()) ()) (= (Var 2 s) (ArrayRef 2 s [((BinOp (ConstantInteger 1 (Integer 4 [])) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()) (BinOp (ConstantInteger 2 (Integer 4 [])) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()) ())] (Character 1 -2 () []) ()) ()) (= (Var 2 s) (ArrayRef 2 s [(() () ())] (Character 1 -2 () []) ()) ()) (= (Var 2 s) (ArrayRef 2 s [(() () (BinOp (UnaryOp USub (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) (ConstantInteger -1 (Integer 4 []))) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()))] (Character 1 -2 () []) ()) ()) (= (Var 2 s) (ArrayRef 2 s [(() () (BinOp (ConstantInteger 2 (Integer 4 [])) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()))] (Character 1 -2 () []) ()) ()) (= (Var 2 s) (ArrayRef 2 s [((BinOp (ConstantInteger 1 (Integer 4 [])) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()) (BinOp (ConstantInteger 88 (Integer 4 [])) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()) (BinOp (ConstantInteger 1 (Integer 4 [])) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()))] (Character 1 -2 () []) ()) ()) (= (Var 2 s) (ArrayRef 2 s [((BinOp (ConstantInteger 1 (Integer 4 [])) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()) (BinOp (ConstantInteger 1 (Integer 4 [])) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()) (BinOp (UnaryOp USub (ConstantInteger 4 (Integer 4 [])) (Integer 4 []) (ConstantInteger -4 (Integer 4 []))) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()))] (Character 1 -2 () []) ()) ()) (= (Var 2 s) (ArrayRef 2 s [((BinOp (UnaryOp USub (ConstantInteger 89 (Integer 4 [])) (Integer 4 []) (ConstantInteger -89 (Integer 4 []))) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()) () (BinOp (ConstantInteger 4 (Integer 4 [])) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()))] (Character 1 -2 () []) ()) ()) (= (Var 2 s) (ArrayRef 2 s [((BinOp (UnaryOp USub (ConstantInteger 3 (Integer 4 [])) (Integer 4 []) (ConstantInteger -3 (Integer 4 []))) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()) (BinOp (UnaryOp USub (ConstantInteger 3 (Integer 4 [])) (Integer 4 []) (ConstantInteger -3 (Integer 4 []))) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()) (BinOp (UnaryOp USub (ConstantInteger 3 (Integer 4 [])) (Integer 4 []) (ConstantInteger -3 (Integer 4 []))) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()))] (Character 1 -2 () []) ()) ()) (= (Var 2 s) (ArrayRef 2 s [((BinOp (ConstantInteger 2 (Integer 4 [])) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()) (BinOp (ConstantInteger 3 (Integer 4 [])) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()) ())] (Character 1 -2 () []) ()) ()) (= (Var 2 i) (ArrayRef 2 A [(() (BinOp (ConstantInteger 0 (Integer 4 [])) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()) ())] (Integer 4 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 5 (Integer 4 [])))]) ()) ()) (= (Var 2 B) (ArrayRef 2 A [((BinOp (ConstantInteger 1 (Integer 4 [])) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()) (BinOp (ConstantInteger 3 (Integer 4 [])) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()) ())] (Integer 4 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 5 (Integer 4 [])))]) ()) ()) (= (Var 2 B) (ArrayRef 2 A [((BinOp (ConstantInteger 1 (Integer 4 [])) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()) (BinOp (ConstantInteger 2 (Integer 4 [])) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()) (BinOp (ConstantInteger 3 (Integer 4 [])) Add (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) () ()))] (Integer 4 [((ConstantInteger 1 (Integer 4 [])) (ConstantInteger 5 (Integer 4 [])))]) ()) ())] Source Public Implementation () .false. .false.)}) [])
